### ISSUE TITLE
Truffleruby claims to be Ruby-2.4.4 compatible but misses Queue#close

### ISSUE THREAD

**Issue by larskanis** [oracle/truffleruby#1442]
`Queue#close` was added to Ruby-2.3 and is already present in JRuby. But it fails on Truffleruby with
```ruby
> Queue.new.close
NoMethodError: undefined method `close' for #<Queue:0x2ae>:Queue
```
Tested on:
```
$ ruby -v
truffleruby 1.0.0-rc6, like ruby 2.4.4, GraalVM CE Native [x86_64-linux]
```

**Comment by chrisseaton** [comment]
When we say we claim compatibility with 2.4.4 we mean that when we implement behaviour we do so aiming to be compatible with 2.4.4, rather than with some other version of Ruby, and we aim to implement all the methods in that version. It doesn't mean that we're exactly compatible with that version of Ruby yet.

I think this is what JRuby mean as well - they aren't exactly compatible with 2.5.0 either.

We don't implement `#close` or `#closed` at the moment, that's true. That's documented and tracked by them being tagged in the specifications.

https://github.com/oracle/truffleruby/tree/4362288c89ce65e814da69cec5bad1693cbabfaf/spec/tags/library/thread

In looking into these methods I also found some inconsistencies in the specifications which all Ruby implementations share, which may have not made it clear we were missing new core library methods.

Then I also found a TruffleRuby bug #1443 around constants which made it even more confusing.

And finally I found that the queue MRI tests are excluded and don't need to be anymore.

This is all fixed in PR #1444.

**Comment by chrisseaton** [comment]
This is fixed in master and should be in RC10. Let me know if you want to know how to build yourself in order to get it now. We went down into a bit of a rabbit-hole about testing locking, and fast-paths for locking, so this was a very helpful issue, thank you very much.
